src/codegen/doc/mod.rs

```diff
@@ -0,0 +1,255 @@
+
+    try!(writeln!(w, "{}src/auto/{}.rs", FILE, class.name.to_lowercase()));
+    if class.doc.is_some() || class.doc_deprecated.is_some() {
+        try!(writeln!(w, "{}{}", MOD_COMMENT, ty));
+    }
+
+    if let Some(ref class_doc) = class.doc {
+        try!(write_lines(w, &class_doc, &tabs));
+    }
+    if let Some(ref class_doc) = class.doc_deprecated {
+        try!(writeln!(w, "{}{}", tabs, DEPRECATED));
+        try!(write_lines(w, &class_doc, &tabs));
+    }
+
+    for function in class.functions.iter() {
+        try!(create_fn_doc(w, &function, Some(Box::new(ty.clone()))));
+    }
+    for child_id in class.children.iter() {
+        if let &LType::Class(ref child) = lib.type_(*child_id) {
+            try!(create_class_doc(w, &child, None, lib));
+        }
+    }
+    Ok(())
+}
+
+fn create_trait_doc(w: &mut Write, trait_: &Interface, parent: Option<Box<TypeStruct>>) -> Result<()> {
+    let tabs : String = primitives::tabs(compute_ident(&parent));
+    let mut ty = trait_.convert();
+    ty.parent = parent;
+
+    if trait_.doc.is_some() || trait_.doc_deprecated.is_some() {
+        try!(writeln!(w, "{}{}", MOD_COMMENT, ty));
+    }
+
+    if let Some(ref trait_doc) = trait_.doc {
+        try!(write_lines(w, &trait_doc, &tabs));
+    }
+    if let Some(ref trait_doc) = trait_.doc_deprecated {
+        try!(writeln!(w, "{}{}", tabs, DEPRECATED));
+        try!(write_lines(w, &trait_doc, &tabs));
+    }
+
+    for function in trait_.functions.iter() {
+        try!(create_fn_doc(w, &function, Some(Box::new(ty.clone()))));
+    }
+    Ok(())
+}
+
+fn create_enum_doc(w: &mut Write, enum_: &Enumeration, parent: Option<Box<TypeStruct>>) -> Result<()> {
+    let ident = compute_ident(&parent);
+    let mut ty = enum_.convert();
+    ty.parent = parent;
+    let tabs : String = primitives::tabs(ident);
+
+    if enum_.doc.is_some() || enum_.doc_deprecated.is_some() {
+        try!(writeln!(w, "{}{}", MOD_COMMENT, ty));
+    }
+
+    if let Some(ref enum_doc) = enum_.doc {
+        try!(write_lines(w, &enum_doc, &tabs));
+    }
+    if let Some(ref enum_doc) = enum_.doc_deprecated {
+        try!(writeln!(w, "{}{}", tabs, DEPRECATED));
+        try!(write_lines(w, &enum_doc, &tabs));
+    }
+
+    let tabs : String = primitives::tabs(ident + 1);
+    for member_doc in enum_.members.iter() {
+        let mut sub_ty : TypeStruct = member_doc.convert();
+
+        if let Some(ref m_doc) = member_doc.doc {
+            sub_ty.parent = Some(Box::new(ty.clone()));
+            try!(writeln!(w, "{}{}", MOD_COMMENT, sub_ty));
+            try!(write_lines(w, &m_doc, &tabs));
+        }
+    }
+    Ok(())
+}
+
+fn create_fn_doc(w: &mut Write, fn_: &Function, parent: Option<Box<TypeStruct>>) -> Result<()> {
+    let tabs : String = primitives::tabs(compute_ident(&parent) + 1);
+
+    let mut docs = match fn_.doc {
+        Some(ref d) => format!("{}/// {}\n", tabs, d.split("\n").collect::<Vec<&str>>().join(&format!("\n{}/// ", tabs))),
```

Comment:
I'd prefer this too was done straightforwardly without collecting into `docs` (not to mention that this function bypasses `write_lines`).
At worst you'll have empty `MOD_COMMENT` line for a couple of functions, which should not have a noticeable effect on stripper.